### DISCUSSION TITLE
LINQ explained with sketches

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#15] [General]
# LINQ explained with sketches

Often times it is easier to have a nice illustration at hand, which explains you things the easy way. So let's do this for a lot of LINQ operations like Where, Select and friends.
Of course a small explanation will be attached as well.

https://steven-giesel.com/blogPost/d65c5411-a69b-489f-b73f-18ce0ed8678d

**Comment by michaelburns** [comment]
Great post, love this!

**Reply by linkdotnet** [reply]
Thanks @michaelburns 

There will be at least 2 more parts ;) So I will regularly update that blog post.
Teaser: The next one comes out on Wednesday ;)

**Reply by bertusviljoen** [reply]
Very cool!

**Comment by dimohy** [comment]
Wow! This is a crazy post. I was very impressed!

**Reply by linkdotnet** [reply]
There will be more ;) Will update the post regulary. Thanks for the supportive words @dimohy

**Comment by muirehenwedd** [comment]
Delightful article! Please check the illustration for MaxBy(), only one element must be returned.

**Reply by linkdotnet** [reply]
Hupps. You are absolutely right. Will be corrected in a second ;) Thanks for reporting.

**Reply by linkdotnet** [reply]
Corrected it. Can take a few minutes to take effect due to caching policies.

**Comment by bpkinez** [comment]
Great illustrations! I think in ***FirstOrDefault*** and ***SingleOrDefault*** methods first element in parentheses should be with blue border or any other border color that is different from input list border colors!

Thanks!

**Reply by linkdotnet** [reply]
Thanks. Very fair point. Will correct that.

**Reply by linkdotnet** [reply]
Corrected. Will take a few minutes for the cache to be invalidated.

**Comment by sam03041993** [comment]
This is very nice article it is really helpful, Keep up the good work !.
